var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -82,6 +82,9 @@ class Opencv(CMakePackage, CudaPackage):
     variant('videoio', default=True, description='videoio module')
 
     # Optional 3rd party components
+    variant('cuda', default=False,
+            description='Activates support for CUDA, \
+                         essentially turns on +contrib')
```

Comment:
This variant is already added by the `CudaPackage` base class, are you overriding it just to change the description?

Reply:
yep. all the contrib stuff comes along for the ride if you do +cuda, so felt we should let the user know - is there a better way to do that?

Reply:
Nope, that's the only way. Although, is it possible to build `+cuda~contrib` or `~cuda+contrib`? If not, we should probably add a `conflicts(...)` statement.

Reply:
I'd really prefer to help the user out by flipping contrib on when they set cuda, but I can also see why that's (unfortunately) complicated.